src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/Html/HtmlHelper.cs

```diff
@@ -748,7 +749,18 @@ protected virtual string GenerateDisplayText(ModelExplorer modelExplorer)
             if (tagBuilder != null)
             {
                 tagBuilder.TagRenderMode = TagRenderMode.StartTag;
-                tagBuilder.WriteTo(ViewContext.Writer, _htmlEncoder);
+
+                // As a perf optimization, we can buffer this output rather than writing it
+                // out character by character.
+                var htmlWriter = ViewContext.Writer as HtmlTextWriter;
+                if (htmlWriter == null)
+                {
+                    tagBuilder.WriteTo(ViewContext.Writer, _htmlEncoder);
+                }
+                else
+                {
+                    htmlWriter.Write(tagBuilder);
+                }
```

Comment:
I'm not sold on that being a better choice. I don't want to put special casey stuff in the `IHtmlContent` implementations, because `HtmlTextWriter` is an mvc.razor detail.

Will address the same special case in `GenerateRouteForm`